packages/next/src/server/app-render/app-render.tsx

```diff
@@ -464,18 +465,28 @@ async function renderToHTMLOrFlightImpl(
   const capturedErrors: Error[] = []
   const allCapturedErrors: Error[] = []
   const isNextExport = !!renderOpts.nextExport
+  const { staticGenerationStore, requestStore } = baseCtx
+  const isStaticGeneration = staticGenerationStore.isStaticGeneration
+  // when static generation fails during PPR, we log the errors separately. We intentionally
+  // silence the error logger in this case to avoid double logging.
+  const silenceStaticGenerationErrors = renderOpts.ppr && isStaticGeneration
```

Comment:
should we put this logic inside the error logger itself?

Reply:
I don't think we want to expose renderOpts & staticGenerationStore to `createErrorHandler` since its interface is already getting a bit unruly, but I can pass this through as a `silenceLogger` variable